### PR TITLE
Issues 1065, 1073, 1074, 1142

### PR DESCRIPTION
**Pull request by lstochlak**
Contains fixes for issues 1065, 1073, 1074, 1142

### CODE REVIEW

**Comment by lstochlak** [comment]
The build fails on GitLab because it takes NMF from dev branch but this version needs changes from Issue-1142 branch.

**Review by CesarCoelho — COMMENTED** [review]
This looks interesting... Is it for the space side?
Can you tell me the startup time before the optimization and after it?
Thanks

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+
+/**
+ * Entity holding lat archive synchronization time for provider.
```
Comment (CesarCoelho) [inline ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java]: "lat" -> "last"

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    private String domain;
+
+    @Column(name = "last_sync", nullable = false)
+    private Long lastSync;
+
+    public LastArchiveSync()
+    {
+    }
+
+    public LastArchiveSync(String providerUri, String domain)
```
Comment (CesarCoelho) [inline ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java:73]: Please also add the doc comments for the javadocs to all the methods you create

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.persistence.EntityManager;
+
+/**
+ * Archive helper class.
+ *
+ * @author Lukasz.Stochlak
+ */
+public class ArchiveHelper
```
Comment (CesarCoelho) [inline ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java]: Can you give a better name to this class?
I believe there is already a class called either ArchiveHelper or HelperArchive.
To me this does not seem like an helper but a new mechanism to find the last sync.
Maybe you should call it: LastSyncHelper or something similar?

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            Logger.getLogger(ArchiveHelper.class.getName()).log(Level.SEVERE, null, e);
+            result = null;
+        }
+
+        emAvailability.release();
+
+        return result;
+    }
+
+    public synchronized boolean saveLastArchiveSync(LastArchiveSync lastArchiveSync)
```
Comment (CesarCoelho) [inline ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java]: "Save" gives the feeling that the state will be stored in a file.
Maybe the method should be called: persistLastArchiveSync   ?

**Comment by lstochlak** [comment]
Before optimizations for 20k parameter definitions I had around 80 seconds for initialization, after optimization I have around 4 s. For details see description https://gitlab.com/esa/NMF/nmf-issues/-/issues/1074#note_683339185

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java (hunk 2 above)
Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java:73]: Done

Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java:73]: Done

Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java:73]: done

Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java:73]: Done

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java (hunk 1 above)
Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/entities/LastArchiveSync.java]: Done

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java (hunk 3 above)
Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java]: Done

File: ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java (hunk 4 above)
Comment (lstochlak) [reply ground-mo-proxy/src/main/java/esa/mo/nmf/groundmoproxy/helper/ArchiveHelper.java]: Done

**Pull request merged**